packages/js/data/src/product-variations/actions.ts

```diff
@@ -44,3 +45,44 @@ export const generateProductVariations = function* ( idQuery: IdQuery ) {
 		throw error;
 	}
 };
+
+export function batchUpdateProductVariationsError(
+	key: IdType,
+	error: unknown
+) {
+	return {
+		type: TYPES.BATCH_UPDATE_VARIATIONS_ERROR as const,
+		key,
+		error,
+		errorType: 'BATCH_UPDATE_VARIATIONS',
+	};
+}
+
+export function* batchUpdateProductVariations(
+	idQuery: IdQuery,
+	data: BatchUpdateRequest
+) {
+	const urlParameters = getUrlParameters(
+		WC_PRODUCT_VARIATIONS_NAMESPACE,
+		idQuery
+	);
+
+	try {
+		const result: BatchUpdateResponse = yield apiFetch( {
+			path: getRestPath(
+				`${ WC_PRODUCT_VARIATIONS_NAMESPACE }/batch`,
+				{},
+				urlParameters
+			),
+			method: 'POST',
+			data,
+		} );
+
+		return result;
```

Comment:
Should we introduce a `batchUpdateProductVariationsSuccess` action to update the state with the newly updated records?

It seems like they are [returned in the response](https://woocommerce.github.io/woocommerce-rest-api-docs/#batch-update-product-variations), but I haven't checked this yet.

Reply:
@joshuatf 
Yes, @louwie17 and I were talking about this and since that is only scoped to this issue the idea is that we can tackle it in a different issue and make this action more generic so other datastore can take advantage of this implementation. Also we need to think about how to implement this more in details because the response contains different lists of items depending on which action is requested `create`, `update` and `delete` which means the way we update the store changes also according the prev actions. Another think is that we should not update the list of items for a key (that was built given a set of params) by the moment the batch update has hapened because at that moment we cannot create the same cache key it was used for listing the items with pagination and sorting for example. Anyway it look like a work for a different issue.